### PR TITLE
schema: add scylla specific options to schema description

### PR DESCRIPTION
**Pull request by Jadw1**
Add `paxos_grace_seconds`, `tombstone_gc`, `cdc` and `synchronous_updates` options to schema description.

Fixes: #12389
Fixes: https://github.com/scylladb/scylla-enterprise/issues/2979

### CODE REVIEW

**Comment by Jadw1** [comment]
@bhalevy let me know if I missed any option

File: schema/schema.cc [hunk 1]
```diff
@@ -906,8 +917,15 @@ std::ostream& schema::describe(replica::database& db, std::ostream& os, bool wit
     os << "\n    AND memtable_flush_period_in_ms = " << memtable_flush_period();
     os << "\n    AND min_index_interval = " << min_index_interval();
     os << "\n    AND read_repair_chance = " << read_repair_chance();
-    os << "\n    AND speculative_retry = '" << speculative_retry().to_sstring() << "';";
-    os << "\n";
+    os << "\n    AND speculative_retry = '" << speculative_retry().to_sstring() << "'";
```
Comment (bhalevy) [inline schema/schema.cc:920]: > Add `speculative_retry`, `paxos_grace_seconds`, `cdc` and `synchronous_updates`
options to schema description.

It looks like `speculative_retry` was already described.

**Comment by bhalevy** [comment]
> @bhalevy let me know if I missed any option

Based on https://github.com/scylladb/scylladb/blob/master/cql3/statements/cf_prop_defs.cc:
`paxos_grace_seconds`

Also, please also examine `operator<<(std::ostream& os, const schema& s)` to see if it prints the same information as `schema::describe()`.

**Comment by Jadw1** [comment]
> Based on https://github.com/scylladb/scylladb/blob/master/cql3/statements/cf_prop_defs.cc: `paxos_grace_seconds`

`paxos_grace_seconds` is present in a describe.
 
> Also, please also examine `operator<<(std::ostream& os, const schema& s)` to see if it prints the same information as `schema::describe()`.

`schema::describe()` doesn't print: `tombstoneGcOptions`, `keyValidator`, `minCompactionThreshold`, `maxCompactionThreshold` and it prints `cdc` only if `cdc_options().enabled()`

`operator<<(...)` doesn't print: `paxos_grace_seconds`, `crc_check_chance`


I've updated the test to include added options, but the CI won't pass because there is sth wrong with boost test's cql env.
1. Creating table with specified `paxos_grace_seconds` doesn't take any effect. The value of `std::optional<int32_t> schema::raw_schema::_paxos_grace_seconds` is `std::nullopt_t`
2. The test cannot read `speculative_retry` because `db::find_tag()` segfaults. The entry from schema's extensions is found but casting `schema_extension` to `tags_extension` fails. The function uses `static_pointer_cast` (thus the segfault) and when I've changed it to dynamic cast the result pointer is null. But the original pointer from extensions map is not null.

**Comment by Jadw1** [comment]
Fixed cql-test env by simply adding `db::extensions` to `cql_test_config`.

Added `tombstone_gc`: https://github.com/scylladb/scylla-enterprise/issues/2979

**Review by bhalevy — APPROVED** [review]

**Comment by Jadw1** [comment]
@bhalevy can we merge this?

**Comment by bhalevy** [comment]
> @bhalevy can we merge this?

It's not up to me...

@scylladb/scylla-maint please merge

**Pull request closed without merging**